client/state/site-settings/reducer.js

```diff
@@ -64,7 +66,21 @@ export const items = createReducer( {}, {
 			...state[ siteId ],
 			...settings
 		}
-	} )
+	} ),
+	[ MEDIA_DELETE ]: ( state, { siteId, mediaIds } ) => {
```

Comment:
If I understand correctly, you're saying that we always rely on the `site` object to retrieve the siteIcon. Thus, we should not worry about resetting it on the siteSettings? Is this assumption correct? Maybe we should consider dropping the selector fallback then, since the `site-settings` value could be incorrect (even if in the current workflow, it can never happen since the site object will be available in this case) ?